hibernate-reactive-core/src/main/java/org/hibernate/reactive/mutiny/impl/MutinySessionFactoryImpl.java

```diff
@@ -34,37 +38,54 @@
 public class MutinySessionFactoryImpl implements Mutiny.SessionFactory {
 
 	private SessionFactoryImpl delegate;
+	private final VertxInstance vertxInstance;
 
 	public MutinySessionFactoryImpl(SessionFactoryImpl delegate) {
 		this.delegate = delegate;
+		vertxInstance = delegate.getServiceRegistry().getService(VertxInstance.class);
+	}
+
+	<T> Uni<T> uni(Supplier<CompletionStage<T>> stageSupplier) {
+		return Uni.createFrom().completionStage(stageSupplier)
+				.runSubscriptionOn( runnable -> {
+					Context context = vertxInstance.getVertx().getOrCreateContext();
+					if ( Vertx.currentContext() == context ) {
+						runnable.run();
+					}
+					else {
+						context.runOnContext( x -> runnable.run() );
+					}
+				} );
```

Comment:
Is this what you mean @cescoffier?

Reply:
exactly. If we are already in the right context, just run the task immediately.